app/sidekiq/central_mail/submit_saved_claim_job.rb

```diff
@@ -11,38 +11,69 @@ class SubmitSavedClaimJob
     include SentryLogging
 
     FOREIGN_POSTALCODE = '00000'
+    STATSD_KEY_PREFIX = 'worker.central_mail.submit_saved_claim_job'
 
-    sidekiq_options retry: false
+    # Sidekiq has built in exponential back-off functionality for retries
+    # A max retry attempt of 14 will result in a run time of ~25 hours
+    RETRY = 14
+
+    sidekiq_options retry: RETRY
 
     class CentralMailResponseError < StandardError
     end
 
+    sidekiq_retries_exhausted do |msg, _ex|
+      Rails.logger.send(
+        :error,
+        "Failed all retries on CentralMail::SubmitSavedClaimJob, last error: #{msg['error_message']}"
+      )
+      StatsD.increment("#{STATSD_KEY_PREFIX}.exhausted")
+    end
+
+    # Performs an asynchronous job for submitting a saved claim to central mail service
+    #
+    # @param saved_claim_id [Integer] the claim id
+    #
     def perform(saved_claim_id)
       PensionBurial::TagSentry.tag_sentry
-      @claim = SavedClaim.find(saved_claim_id)
-      @pdf_path = process_record(@claim)
-
-      @attachment_paths = @claim.persistent_attachments.map do |record|
-        process_record(record)
-      end
+      @saved_claim_id = saved_claim_id
+      log_message_to_sentry('Attempting CentralMail::SubmitSavedClaimJob', :info, generate_sentry_details)
 
-      response = CentralMail::Service.new.upload(create_request_body)
+      response = send_claim_to_central_mail(saved_claim_id)
       log_cmp_response(response) if @claim.is_a?(SavedClaim::VeteranReadinessEmploymentClaim)
-      File.delete(@pdf_path)
-      @attachment_paths.each { |p| File.delete(p) }
 
       if response.success?
         update_submission('success')
+        log_message_to_sentry('CentralMail::SubmitSavedClaimJob succeeded', :info, generate_sentry_details)
 
         @claim.send_confirmation_email if @claim.respond_to?(:send_confirmation_email)
       else
-        raise CentralMailResponseError
+        raise CentralMailResponseError, response.to_s
       end
-    rescue
+    rescue => e
       update_submission('failed')
+      log_message_to_sentry(
+        'CentralMail::SubmitSavedClaimJob failed, retrying...', :warn, generate_sentry_details(e)
+      )
       raise
     end
 
+    def send_claim_to_central_mail(saved_claim_id)
+      @claim = SavedClaim.find(saved_claim_id)
```

Comment:
Would using ||= be useful here, and for all the instance variable instantiations? I'm not sure if this info persists between retry attempts, but if it does persist between retries, this could help reduce redundant DB calls, running of logic, etc. if it was successful the first time. If not, then please ignore my suggestion.

Reply:
I took a look into this, and it looks like the instance variables don't persist between retry attempts.